pkg/controllers/collaset/synccontrol/sync_control.go

```diff
@@ -348,37 +390,41 @@ func extractAvailableContexts(diff int, ownedIDs map[int]*appsv1alpha1.ContextDe
 	return availableContexts
 }
 
-func (sc *RealSyncControl) Update(cls *appsv1alpha1.CollaSet, podWrapers []*collasetutils.PodWrapper, revisions []*appsv1.ControllerRevision, updatedRevision *appsv1.ControllerRevision, ownedIDs map[int]*appsv1alpha1.ContextDetail, newStatus *appsv1alpha1.CollaSetStatus) (bool, time.Duration, error) {
-	logger := sc.logger.WithValues("collaset", commonutils.ObjectKeyString(cls))
-	var recordedRequeueAfter time.Duration
+func (r *RealSyncControl) Update(
+	ctx context.Context,
+	cls *appsv1alpha1.CollaSet,
+	resources *collasetutils.RelatedResources,
+	podWrappers []*collasetutils.PodWrapper,
+	ownedIDs map[int]*appsv1alpha1.ContextDetail,
+) (bool, *time.Duration, error) {
+
+	logger := r.logger.WithValues("collaset", commonutils.ObjectKeyString(cls))
+	var recordedRequeueAfter *time.Duration
 	// 1. scan and analysis pods update info
-	podUpdateInfos := attachPodUpdateInfo(podWrapers, revisions, updatedRevision)
+	podUpdateInfos := attachPodUpdateInfo(podWrappers, resources)
 
 	// 2. decide Pod update candidates
 	podToUpdate := decidePodToUpdate(cls, podUpdateInfos)
 
 	// 3. prepare Pods to begin PodOpsLifecycle
 	podCh := make(chan *PodUpdateInfo, len(podToUpdate))
-	for _, podInfo := range podToUpdate {
-		if podInfo.IsUpdatedRevision {
+	for i, podInfo := range podToUpdate {
+		if podInfo.IsUpdatedRevision && !podInfo.PodDecorationChanged {
```

Comment:
We may move `!podInfo.PodDecorationChanged` into `decidePodToUpdate(cls, podUpdateInfos)` to decide, together with CollaSet, which Pod need to update.